### PR TITLE
Link patient invoice payment

### PR DESCRIPTION
**Pull request by lomamech**
Remake of pull request in Branch Master

### CODE REVIEW

File: client/src/js/services/PatientInvoiceService.js [hunk 1]
```diff
@@ -120,6 +120,7 @@ function PatientInvoiceService(Modal, util, Session, Api) {
       { field: 'service_id', displayName: 'FORM.LABELS.SERVICE' },
       { field: 'user_id', displayName: 'FORM.LABELS.USER' },
       { field: 'reference', displayName: 'FORM.LABELS.REFERENCE' },
+      { field: 'debtor_uuid', displayName: 'FORM.LABELS.CLIENT' },
```
Comment (jniles) [inline client/src/js/services/PatientInvoiceService.js:123]: Do you need this?  We shouldn't be showing `uuid`s to the user...

File: client/src/partials/patient_invoice/registry/search.modal.html [hunk 2]
```diff
@@ -39,12 +39,30 @@
...
         </span>
-        <input type="text" class="form-control" name="reference"ng-model="ModalCtrl.params.reference">
+        <input type="text" class="form-control" name="reference" ng-model="ModalCtrl.params.reference">
         <div class="help-block" ng-messages="ModalForm.reference.$error" ng-show="ModalForm.$submitted">
           <div ng-messages-include="partials/templates/messages.tmpl.html"></div>
         </div>
       </div>
 
+      <!-- client -->
+      <div class="form-group">
```
Comment (jniles) [inline client/src/partials/patient_invoice/registry/search.modal.html]: I know from experience that this ui-select will crash some of the computers at Vanga ... listing every debtor in a dropdown is _far_ too big a task.   I propose you remove this `ui-select` altogether.

File: client/src/partials/patients/templates/action.cell.html [hunk 3]
```diff
@@ -20,5 +20,16 @@
         <span class="fa fa-user"></span> <span translate>PATIENT_REGISTRY.CARD</span>
       </a>
     </li>
+    <li class="divider"></li>
+    <li>
+      <a data-method="card"  ui-sref="invoiceRegistry({ filters : { debtor_uuid : row.entity.debtor_uuid }, display : {debtor_uuid : row.entity.display_name}})" href>
```
Comment (jniles) [inline client/src/partials/patients/templates/action.cell.html:25]: :+1:

File: client/src/partials/patients/templates/action.cell.html [hunk 4]
```diff
@@ -20,5 +20,16 @@
...
+        <span class="fa fa-file-text-o"></span> <span translate>PATIENT_REGISTRY.INVOICES</span>
+      </a>
+    </li>
+    <li>
+      <a data-method="card" ui-sref="cashRegistry({ filters : { debtor_uuid : row.entity.debtor_uuid }, display : {debtor_uuid : row.entity.display_name}})" href>
```
Comment (jniles) [inline client/src/partials/patients/templates/action.cell.html:30]: :+1:

**Review by jniles — COMMENTED** [review]
@lomamech this is pretty good.

That `ui-select` will crash some of our client machines here - loading 3000+ debtors into a `ui-select` isn't good for performance.  If you can remove it, this can be merged.

File: client/src/js/services/PatientInvoiceService.js (hunk 1 above)
Comment (lomamech) [reply client/src/js/services/PatientInvoiceService.js:123]: The debtor_uuid property was to add to formatFilterParameters to allow the FORM.LABELS.CLIENT display to the case the filtering is performed relative to the Debitor

Comment (lomamech) [reply client/src/js/services/PatientInvoiceService.js:123]: ui-select Allows you to search for debtors in
- partials / patient_invoice / registry / search.modal.html
- partials / cash / payments / templates / search.modal.html

Should I delete it in this two Modals

Comment (jniles) [reply client/src/js/services/PatientInvoiceService.js:123]: Yes, that would be best as well.

**Comment by jniles** [comment]
Travis passes, but doesn't report it to Github.  I'm merging this.

**Pull request merged**